### PR TITLE
Add leap KDE to SLE migration with yast2 migration

### PR DESCRIPTION
**Pull request by hjluo**
In leap KDE to SLE yast2 migration, we need to disable the
screensaver during the migration. And in snapper_rollback
when roll back to leap, we don't need to check the display manager
for the kde and gnome desktop.

- Related ticket: https://progress.opensuse.org/issues/91263
- Needles: N/A
- Verification run: 
  https://openqa.nue.suse.com/tests/5887332
  https://openqa.nue.suse.com/tests/5888023
  https://openqa.nue.suse.com/tests/5888041
  http://openqa.nue.suse.com/tests/5893937 from SLES15 gnome
  https://openqa.opensuse.org/tests/1711790 from opensuse
  https://openqa.nue.suse.com/tests/5894049 leap to sle gnome

### CODE REVIEW

File: tests/migration/online_migration/yast2_migration.pm [hunk 1]
```diff
@@ -237,7 +237,7 @@ sub run {
     assert_screen "yast2-migration-upgrading", 120;
 
     # start migration
-    my $timeout = 7200;
+    my $timeout = 600;
```
Comment (okurz) [inline tests/migration/online_migration/yast2_migration.pm:240]: I assume you only did that for testing?

**Review by okurz — CHANGES_REQUESTED** [review]

File: tests/migration/online_migration/yast2_migration.pm (hunk 1 above)
Comment (hjluo) [reply tests/migration/online_migration/yast2_migration.pm:240]: yes.

**Comment by hjluo** [comment]
@okurz Hi Oliver can you help to review it again? thanks!

File: tests/boot/snapper_rollback.pm [hunk 2]
```diff
@@ -22,7 +22,7 @@ use version_utils;
 
 sub run {
     my ($self) = @_;
-    if (is_leap_migration && check_var('DESKTOP', 'gnome')) {
+    if (is_leap_migration && (check_var('DESKTOP', 'gnome') || check_var('DESKTOP', 'kde'))) {
```
Comment (okurz) [inline tests/boot/snapper_rollback.pm:25]: This change seems like something that should be in it's own commit and please provide according openSUSE validation tests

Comment (hjluo) [reply tests/boot/snapper_rollback.pm:25]: @okurz I've changed the commit message, please take a look again. thanks.

File: tests/migration/online_migration/yast2_migration.pm [hunk 3]
```diff
@@ -112,8 +112,8 @@ sub run {
         mouse_hide(1);
         assert_screen 'generic-desktop';
 
+        turn_off_screensaver();
```
Comment (lemon-suse) [inline tests/migration/online_migration/yast2_migration.pm:115]: It is better to run more verification log such as online migration from SLES15*, so is for opensuse migration.

Comment (hjluo) [reply tests/migration/online_migration/yast2_migration.pm:115]: http://openqa.nue.suse.com/tests/5893937  from SLES15
https://openqa.opensuse.org/t1711790        opensuse

File: tests/boot/snapper_rollback.pm (hunk 2 above)
Comment (okurz) [reply tests/boot/snapper_rollback.pm:25]: I crosschecked and it really looks like snapper_rollback is not anymore tested within openSUSE which I see as a serious issue. I strongly recommend to fix that first by investigating how the module got lost from testing and then bring it back.

**Review by okurz — CHANGES_REQUESTED** [review]

File: tests/boot/snapper_rollback.pm (hunk 2 above)
Comment (hjluo) [reply tests/boot/snapper_rollback.pm:25]: cause they set INSTALLONLY=1 in opensuse.

Comment (okurz) [reply tests/boot/snapper_rollback.pm:25]: Could be. But still: I strongly recommend to fix that first by investigating how the module got lost from testing and then bring it back.

Comment (hjluo) [reply tests/boot/snapper_rollback.pm:25]: we've checked the code and found:
if (get_var('ROLLBACK_AFTER_MIGRATION') && (snapper_is_applicable())) {
load_rollback_tests();
}
sub snapper_is_applicable {
# run snapper tests on opensuse only for system_performance testsuite
return 0 if (is_opensuse && !get_var('SOFTFAIL_BSC1063638'));
this is the root cause why not run snapper_rollback on opensuse
So it is expected.

Comment (okurz) [reply tests/boot/snapper_rollback.pm:25]: I don't think so. The comment says "run snapper tests on opensuse only for system_performance testsuite" so that "system_performance" testsuite should cover it then but it apparently isn't, see https://openqa.opensuse.org/tests/latest?arch=x86_64&distri=opensuse&flavor=DVD&machine=64bit&test=system_performance&version=Tumbleweed

I suggest to be a bit careful with a statement like "So it is expected." as humans make mistakes so likely that is not really expected but potentially an unexpected regression in test coverage

Comment (hjluo) [reply tests/boot/snapper_rollback.pm:25]: that'll call load_rollback_tests and in  load_rollback_tests {

    if (get_var('UPGRADE') || get_var('ZDUP')) {
        loadtest "boot/snapper_rollback";
    }
    if (get_var('MIGRATION_ROLLBACK')) {
        loadtest "migration/online_migration/snapper_rollback";
    }
only these settings will load the snapper_rollback. 
that's what I can think of it.

Comment (hjluo) [reply tests/boot/snapper_rollback.pm:25]: Now with adding ROLLBACK_AFTER_MIGRATION, FILESYSTEM=btrfs to test system_performance
and with a branch https://github.com/hjluo/os-autoinst-distri-opensuse/tree/opensuse_rollback
it'll have the snapper_rollbak and case https://openqa.opensuse.org/tests/1716989 PASSED
So can you approve this PR or file another ticket for the issue in OpenSuSE? thanks

**Comment by okurz** [comment]
I already provided my review comments and I do not know if there was a resolution for https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/12375#discussion_r620032303

**Comment by lemon-suse** [comment]
> I already provided my review comments and I do not know if there was a resolution for [#12375 (comment)](https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/12375#discussion_r620032303)

Hi Oliver, thanks for the careful review, as huajian's investigation, the reason that snapper_rollback not loaded on opensuse is for the missed setting of ROLLBACK_AFTER_MIGRATION and FILESYSTEM. I suggest to merge this PR firstly and create another ticket to fix the opensuse issue for snapper_rollback, huajian can fix it later. WDYT?

**Comment by okurz** [comment]
sure, sounds good

**Comment by hjluo** [comment]
Hi Oliver, now I've added a ticket to add snapper_rollback to OpenSUSE. https://progress.opensuse.org/issues/91992
Can you approve this PR first while I can add it with this ticket later? WDYT?

**Review by okurz — APPROVED** [review]
thanks for the ticket. Sorry for the delay. I struggle to remove my change request but I can approve now :)

**Review by coolgw — APPROVED** [review]
